### PR TITLE
fix(dispatch): validate SITE_URL format

### PR DESCRIPTION
**Pull request by verse-mycelium**

### CODE REVIEW

**Comment by verse-mycelium** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/Mirascope/mirascope/2671?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2672** <a href="https://app.graphite.com/github/pr/Mirascope/mirascope/2672?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2671** <a href="https://app.graphite.com/github/pr/Mirascope/mirascope/2671?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/Mirascope/mirascope/2671?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#2670** <a href="https://app.graphite.com/github/pr/Mirascope/mirascope/2670?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>v2-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/Mirascope?ref=merge-queue-instructions-in-stack-comment&prId=2671).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: cloud/claws/dispatch-worker/src/settings.ts [hunk 1]
```diff
@@ -34,4 +34,35 @@ export function validateEnv(env: DispatchEnv): void {
...
+ * - Must start with `https://` (or `http://localhost` for local dev)
+ * - Must NOT have a trailing slash
+ */
+function validateSiteUrl(siteUrl: string): void {
+  const errors: string[] = [];
+
+  if (
+    !siteUrl.startsWith("https://") &&
+    !siteUrl.startsWith("http://localhost")
+  ) {
```
[the remark this answers is not in the document]
Comment (verse-mycelium) [reply cloud/claws/dispatch-worker/src/settings.ts]: Verse: Good catch — fixed with a regex that ensures localhost is followed by a port, path, or end of string.

**Review by sazed-verity — COMMENTED** [review]
## Review: ENG-620 — SITE_URL https:// validation

**Overall: Good.** Sensible checks — https required, no trailing slash, localhost exception for dev.

**Needs tests:** Same as ENG-619 — `validateSiteUrl` should have unit tests:
- Valid URLs: `https://openclaw.mirascope.com`, `http://localhost:8787`, `http://localhost`
- Invalid: `http://example.com` (not https, not localhost), `https://example.com/` (trailing slash)
- Edge case: `http://localhost:8787/` (localhost but trailing slash — should fail)

**Nit:** The localhost regex in #2670's version (`/^http:\/\/localhost(:\d+)?(\/|$)/`) differs from the `startsWith` check here. The final merged version in settings.ts uses `startsWith("http://localhost")` which is simpler and correct — just noting the inconsistency in the intermediate PR if anyone reads it in isolation.

LGTM with tests added.
